src/riak_kv_clusteraae_fsm.erl

```diff
@@ -556,50 +556,62 @@ finish(clean, State=#state{from={raw, ReqId, ClientPid}}) ->
 %% External functions
 %% ===================================================================
 
--spec json_encode_results(query_types(), query_return()) -> iolist().
+-spec json_encode_results(
+    query_types()|dispatched_count, query_return()) -> iolist().
 %% @doc
 %% Encode the results of a query in JSON
 %% Expected this will be called from the webmachine module that needs to
 %% generate the response
 json_encode_results(merge_root_nval, Root) ->
-    RootEnc = base64:encode_to_string(Root),
-    Keys = {struct, [{<<"root">>, RootEnc}]},
-    mochijson2:encode(Keys);
+    Keys = #{<<"root">> => base64:encode_to_string(Root)},
+    thoas:encode_to_iodata(Keys);
 json_encode_results(merge_branch_nval, Branches) ->
-    Keys = {struct, [{<<"branches">>, [{struct, [{<<"branch-id">>, BranchId},
-                                                 {<<"branch">>, base64:encode_to_string(BranchBin)}]
-                                       } || {BranchId, BranchBin} <- Branches]
-                     }]},
-    mochijson2:encode(Keys);
+    Keys =
+        #{<<"branches">> =>
+            lists:map(
+                fun({BranchId, BranchBin}) -> 
+                    #{<<"branch-id">> => BranchId,
+                        <<"branch">> => base64:encode_to_string(BranchBin)}
+                end,
+                Branches)},
+    thoas:encode_to_iodata(Keys);
 json_encode_results(fetch_clocks_nval, KeysNClocks) ->
     encode_keys_and_clocks(KeysNClocks);
 json_encode_results(merge_tree_range, Tree) ->
-    ExportedTree = leveled_tictac:export_tree(Tree),
-    JsonKeys1 = {struct, [{<<"tree">>, ExportedTree}]},
-    mochijson2:encode(JsonKeys1);
+    {struct, 
+        [{<<"level1">>, EncodedL1}, 
+            {<<"level2">>, {struct, EncodedL2}}]} =
+        leveled_tictac:export_tree(Tree),
+    thoas:encode_to_iodata(
+        #{<<"tree">> =>
+            #{<<"level1">> => EncodedL1, <<"level2">> => EncodedL2}});
 json_encode_results(fetch_clocks_range, KeysNClocks) ->
     encode_keys_and_clocks(KeysNClocks);
+json_encode_results(dispatched_count, Count) ->
+    R = #{<<"dispatched_count">> => Count},
+    thoas:encode_to_iodata(R);
 json_encode_results(repl_keys_range, ReplResult) ->
-    R = {struct, [{<<"dispatched_count">>, element(2, ReplResult)}]},
-    mochijson2:encode(R);
+    json_encode_results(dispatched_count, element(2, ReplResult));
 json_encode_results(repair_keys_range, ReplResult) ->
-    R = {struct, [{<<"dispatched_count">>, element(2, ReplResult)}]},
-    mochijson2:encode(R);
+    json_encode_results(repl_keys_range, ReplResult);
```

Comment:
```suggestion
    json_encode_results(dispatched_count, element(2, ReplResult));
```

I like the original, less indirection, better. When reading code as it is now, I need to understand that it goes to dispatched_count in 2 steps